include/staticbufferheap.hpp

```diff
@@ -34,6 +41,10 @@ class StaticBufferHeap {
     return ptr;
   }
 
+  void *memalign(size_t alignment, size_t sz) {
+    return malloc(sz); // FIXME 'alignment' ignored
```

Comment:
Since we are not implementing `free` (that is, it's a NOP), then we can actually implement `memalign` here just in case by allocating a sufficiently-large chunk of memory and then placing the allocated object in the right spot within it. Anyway, this should arguably be using a lock layer around https://github.com/emeryberger/Heap-Layers/blob/master/heaps/top/staticheap.h, right?

Reply:
Good point about just fixing`memalign`.

We don't need a separate lock guard here as long as we invoke `malloc`, since that method has it.

Reply:
Right, I meant that the whole thing could just be a `LockedHeap` around a slightly modified `StaticHeap`.

Reply:
Ah yes. I'll check whether `std::mutex` and `std::lock_guard` are ok to use... after all the difficulties with `std::call_once` and `thread_local` I went all `pthread`. If that works, we'd just need to modify it for alignment (in `malloc` and in `memalign`).

Reply:
If you use `LockedHeap`, you can just use any class with the same interface as `std::mutex`: see https://github.com/emeryberger/Heap-Layers/blob/master/heaps/threads/lockedheap.h.

Reply:
> Since we are not implementing `free` (that is, it's a NOP), then we can actually implement `memalign` here just in case by allocating a sufficiently-large chunk of memory and then placing the allocated object in the right spot within it.

Actually, not exactly, as `getSize` wouldn't work, but we can skip enough in the buffer to arrange for the alignment. Working on it.

Reply:
> If you use `LockedHeap`, you can just use any class with the same interface as `std::mutex`: see https://github.com/emeryberger/Heap-Layers/blob/master/heaps/threads/lockedheap.h.

Could you modify `LockedHeap` to also wrap-lock around `memalign`?  Then I could refactor the locking out of `StaticBufferHeap`.

Reply:
> Could you modify `LockedHeap` to also wrap-lock around `memalign`? Then I could refactor the locking out of `StaticBufferHeap`.

I've opened a PR for that on Heap-Layers.

Reply:
Yep, merged!